platform/mbed_critical.c

```diff
@@ -23,6 +23,21 @@
 #include "platform/mbed_assert.h"
 #include "platform/mbed_toolchain.h"
 
+// if __EXCLUSIVE_ACCESS rtx macro not defined, we need to get this via own-set architecture macros
+#ifndef __EXCLUSIVE_ACCESS
+#ifndef MBED_EXCLUSIVE_ACCESS
+#if ((__ARM_ARCH_7M__      == 1U) || \
```

Comment:
I know this is copying the logic from RTX, but does kind of feel like the test should be the other way around - 0 only if 6M (or 6A). I think we can assume any future architecture will have exclusive access instructions, and we're not backporting to pre-6.

(On the other hand, I'm aware of platforms where the core supported exclusive access instructions, but the memory system lacked a correctly-functioning global monitor so they didn't work properly - so it's worth having the possibility for platform override.)

Reply:
> I know this is copying the logic from RTX, but does kind of feel like the test should be the other way around - 0 only if 6M (or 6A). I think we can assume any future architecture will have exclusive access instructions, and we're not backporting to pre-6.

I could do that. I added a new commit that lists all architecture we are aware of and emits an error if not recognized (or can do warning and default to non-exclusive access, however that does not make much sense?).

Reply:
That error version seems fine. I think it's worth double-checking this sort of code when adding a new arch anyway.